### PR TITLE
feat(remove-cookies): create plugin,tests and registration

### LINKED ISSUE #17: ignore cookie headers from all requests
Although hoodie-server-acocunt does not set any cookies, there are cases where a cookie might be set and cause very hard bugs. Cookies are scoped by host, not including the port number. So if the API from `hoodie-server-account` is accessible at http://localhost:4000/api/account and CouchDB is at http://localhost:5984, CouchDB’s `AuthSession` cookie will be sent to http://localhost:4000/api/account as well.

We don’t proxy requests at this point, but I'd say we should do this anyway for sanity

Simple fix is: remove all cookie headers in all requests. To achieve that, [hapi’s `server.ext`](http://hapijs.com/api#serverextevent-method-options) method can be used to hook into [hapi’s request lifecycle](http://hapijs.com/api#request-lifecycle). Make sure to use the `sandbox: 'plugin'` option on `server.ext`. 

- [ ] Start Pull Request to claim the issue and start discussion
- [ ] write a failing tests for any request sending a cookie
- [ ] fix test
- [ ] review & merge

### PR DESCRIPTION
**Pull request by sebbel**
This PR is still WIP as I have failing tests and I would need some feedback. I am also unsure if I have placed the removeCookie plugin into the right folder.

---

fixes #17

### CODE REVIEW

**Comment by gr2m** [comment]
Close, `response.raw.req.headers.cookie` should be `undefined` not `null`, and the `group.end()` call must go into the same block as `group.test`

```diff
diff --git a/tests/integration/routes/requests/no-cookie-header-test.js b/tests/integration/routes/requests/no-cookie-header-test.js
index 431e4b2..cd468a3 100644
--- a/tests/integration/routes/requests/no-cookie-header-test.js
+++ b/tests/integration/routes/requests/no-cookie-header-test.js
@@ -30,12 +30,13 @@ test('Recieve no cookie on GET Request', function (group) {
 
     group.test('GET no cookie', function (t) {
       server.inject(routeOptions, function (response) {
-        t.is(response.raw.req.headers.cookie, null, 'Header Cookie on GET Request should be null')
+        t.is(response.raw.req.headers.cookie, undefined, 'Header Cookie on GET Request should be null')
         t.end()
       })
     })
+
+    group.end()
   })
-  group.end()
 })
 
 test('Recieve no cookie on POST Request', function (group) {
@@ -50,10 +51,11 @@ test('Recieve no cookie on POST Request', function (group) {
 
     group.test('POST no cookie', function (t) {
       server.inject(routePOSTOptions, function (response) {
-        t.is(response.raw.req.headers.cookie, null, 'Header Cookie on GET Request should be null')
+        t.is(response.raw.req.headers.cookie, undefined, 'Header Cookie on GET Request should be null')
         t.end()
       })
     })
+
+    group.end()
   })
-  group.end()
 })
```

A few more notes:

1. Can you rename `removeCookie.js` to `remove-cookie-header.js`
2. Can you follow the conventions from [plugin/index.js](https://github.com/hoodiehq/hoodie-account-server/blob/master/plugin/index.js#L1-L4), meaning define a `function removeCookieHeader () {}`  like `function hapiAccount () {}` is defined?
3. register the plugin in `plugin/index.js` with `sandbox: 'plugin'` option
4. Move the `no-cookie-header-test.js` one directory up into `tests/integration/routes`

File: tests/integration/routes/requests/no-cookie-header-test.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    cookie: 'testCookie'
+  }
+}
+
+test('Recieve no cookie on GET Request', function (group) {
+  getServer({
+  }, function (error, server) {
+    if (error) {
+      return test('test setup', function (t) {
+        t.error('error')
```
Comment (gr2m) [inline tests/integration/routes/requests/no-cookie-header-test.js]: this should be `t.error(error)`, not `t.error('error')`

**Comment by sebbel** [comment]
Thanks for the help @gr2m :) I've changed the code according to your suggestions and have passing tests now (wohooooo!!)

**Comment by gr2m** [comment]
congrats, looking great :)

3 more things

- can you please revert the whitespace changes in `plugin/index.js` and `tests/integration/utils/get-server.js`
- rebase on latest `master` branch
- squash commits two only two:
  1. `test: remove request cookies`
  2. `fix: remove request cookies`

Happy to help with any of these if you are busy or have any questions

**Comment by sebbel** [comment]
From the looks of it and my poor git Skills, how would i approach these
tasks?

1. git revert get-server? would i need to apply another commit to the
plugin index.js in order to get rid of the white space?
2. rebase as in merge latest master??
3. I never squashed yet :(

thanks again Gregor

Am Dienstag, 12. April 2016 schrieb Gregor Martynus :

**Comment by gr2m** [comment]
no problem at all, `git` is a mystery to all of us, I’m happy to help :)

### Remove whitespace

To remove the whitespace, make an extra commit, the commit message does not matter. double check in the [diff view](https://github.com/hoodiehq/hoodie-account-server/pull/145/files) that all whitespace-only changes have been reverted

### Rebase on master

First, add the original repository (https://github.com/hoodiehq/hoodie-account-server) as an additional git remote, we call it `upstream`

```
git remote add upstream git@github.com:hoodiehq/hoodie-account-server.git
```

No fetch all branches from the `upstream` remote

```
git fetch upstream
```

Now you can rebase your fork on the latest `upstream/master`

```
git rebase upstream/master
```

### Squash commits

There are different ways to do it. In your case the easiest is to reset all commits, and then commit from scratch. First you need to find out the latest commit hash of the branch you want to merge into, which is `upstream/master`. You can do that with

```
$ git merge-base upstream/master HEAD
17d3c118291875701b825dd078077e82441f1284
```

Now taht you have the hash, you can do

```
git reset 17d3c118291875701b825dd078077e82441f1284
```

If you run `git status` now, you will see all the changes you did in your fork.

Now do the two commits from scratch

```
git add tests
git commit -m 'test: remove request cookies'
git add plugin
git commit -m 'fix: remove request cookies'
```

Last put not least, git push -f to your fork

```
git push --force origin master
```

That should be it :)

File: plugin/remove-cookie-header.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+module.exports = removeCookieHeader
+removeCookieHeader.attributes = {
+  name: 'removeCookie'
+}
+
+function removeCookieHeader (server, options, next) {
+  server.ext({
+    type: 'onRequest',
+    method: function (request, reply) {
+      // Change all requests to '/test'
```
Comment (gr2m) [inline plugin/remove-cookie-header.js]: oh sorry, I missed this before. What is this comment about? I think we can remove it?

You can remove the comment, then run `git commit plugin --amend -m 'fix: remove cookie from requests'`, that will update the last commit. Then you can do `git push --force origin master` again

**Comment by gr2m** [comment]
LGTM ping @hoodiehq/maintainers for 2nd review :)

**Comment by HipsterBrown** [comment]
LGTM! Merge away! :clap: 

Great work @sebbel!

**Comment by gr2m** [comment]
ohh yeah :)

**Pull request merged**